### PR TITLE
added not expressions and deprecate not logic

### PR DESCRIPTION
**Pull request by AlessandroMinoccheri**

### CODE REVIEW

File: src/Expression/Logical/Not.php [hunk 1]
```diff
@@ -9,6 +9,9 @@
 use Arkitect\Expression\NegativeDescription;
 use Arkitect\Rules\Violations;
 
+/**
+ * @deprecated deprecated
+ */
 class Not implements Expression
```
Comment (fain182) [inline src/Expression/Logical/Not.php:15]: Why deprecate and not delete? We found that is misleading, so for me it's better remove it.

**Comment by fain182** [comment]
We need to update the README too, I think..

File: src/Expression/Logical/Not.php (hunk 1 above)
Comment (AlessandroMinoccheri) [reply src/Expression/Logical/Not.php:15]: Yes, this is a BC and I tried a different approach before delete it completely. 
I'll remove it completely.

Comment (micheleorselli) [reply src/Expression/Logical/Not.php:15]: As of now Not it is just not working as expected, we should get rid of it

Comment (AlessandroMinoccheri) [reply src/Expression/Logical/Not.php:15]: I have Removed the class `Not`

**Review by micheleorselli — APPROVED** [review]

**Pull request merged**